arviz/data/inference_data.py

```diff
@@ -1394,10 +1394,11 @@ def extend(self, other, join="left"):
                 )
             dataset = getattr(other, group)
             setattr(self, group, dataset)
-            if group.startswith(WARMUP_TAG):
-                self._groups_warmup.append(group)
-            else:
-                self._groups.append(group)
+            if group not in self._groups and self._groups_warmup:
```

Comment:
This line tests if both of these are true

    bool(group not in self._groups)

    bool(self._group_warmup)

Is this what is wanted?

Reply:
i'm not very sure(looking at this PR after a while so lost some context, feel free to correct me) but i think we were trying to ensure no group is repeated twice when we do extend with 'right'. is the group_warmup list also used finally in the inferencedata object returned?